### PR TITLE
Remove csi namespace configuration from controller

### PR DESCRIPTION
**Pull request by nikhilbarge**
### What this PR does / why we need it

This will help remove control of vsphere-csi namespace from t-f addons-manager. CSI will be deployed in namespace configure in vpshere-csi package.

### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR
run e2e tests 
run [e2e tests](https://kscom.svc.eng.vmware.com/job/tkg-release/job/upgrade-vc7/8890) for leagcy cluster 

### Release note

```release-note
CSI will be deployed in namespace configure in vpshere-csi package. addon-controller will not manage vsphere-csi namespace.
```

### CODE REVIEW

File: addons/controllers/csi/vspherecsiconfig_datavalues.go [hunk 1]
```diff
@@ -23,7 +23,6 @@ type DataValuesVSpherePVCSI struct {
 
 type DataValuesVSphereCSI struct {
 	TLSThumbprint         string                          `yaml:"tlsThumbprint"`
-	Namespace             string                          `yaml:"namespace"`
```
[the remark this answers is not in the document]
Comment (nikhilbarge) [reply addons/controllers/csi/vspherecsiconfig_datavalues.go]: as discussed namespace field will not be removed but we will not use it from csi v2.7.1 later, we will deprecate it later releases.

**Comment by alfredthenarwhal** [comment]
> @nikhilbarge:  /test install-vc7
Commit: 0ff5d67

Tests failed! Build no: 4531

**Comment by nikhilbarge** [comment]
//usebom:sandbox/v2.2.0-zshippable/453052567442722630/tkg-compatibility

**Comment by alfredthenarwhal** [comment]
> @nikhilbarge:  /test upgrade-vc7
Commit: 0ff5d67

Tests failed! Build no: 4544

**Comment by nikhilbarge** [comment]
//usebom:sandbox/v2.2.0-zshippable/453162315970280738/tkg-compatibility

**Comment by nikhilbarge** [comment]
//usebom:sandbox/v2.2.0-zshippable/453162315970280738/tkg-compatibility

**Comment by alfredthenarwhal** [comment]
> @nikhilbarge:  /test upgrade-vc7
Commit: 5a71fa8

Tests failed! Build no: 4554

**Comment by alfredthenarwhal** [comment]
> @nikhilbarge:  /test upgrade-vc7
Commit: 5a71fa8

Tests failed! Build no: 4560

**Comment by alfredthenarwhal** [comment]
> @nikhilbarge:  /test upgrade-vc7
Commit: 5a71fa8

Tests failed! Build no: 4566

**Comment by alfredthenarwhal** [comment]
> @nikhilbarge:  /test upgrade-vc7
Commit: 5a71fa8

Tests passed! Build no: 4569

**Review by 12345lcr — APPROVED** [review]

**Pull request merged**